python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -469,6 +535,24 @@ def scale_down(self, num_actors: int) -> int:
 
     # === End of overriding methods of AutoscalingActorPool ===
 
+    def mark_running_actor_as_restarting(self, actor: ray.actor.ActorHandle):
+        """Mark the running actor as restarting.
+
+        Args:
+            actor: The running actor to be marked as restarting.
+        """
+        assert actor in self._running_actors
+        self._running_actors[actor].is_restarting = True
+
+    def mark_actor_as_alive(self, actor: ray.actor.ActorHandle):
+        """Mark the running actor as alive.
+
+        Args:
+            actor: The running actor to be marked as alive.
+        """
+        assert actor in self._running_actors
+        self._running_actors[actor].is_restarting = False
```

Comment:
Let's align the terminology b/w Actor Pool and Ray Core

 - Actor Pool calls all of the actors "running" irrespective whether they are ALIVE or not
 - This doesn't line up with terms of Ray Core where only `ALIVE` actors are considered running

To remediate i suggest following:

 - Rename `ActorRunningState` to become `ActorState`
 - Instead of booleans just persist actual state reported by Ray Core
 - Align param/method names with Core terms